docs/guide/custom-ui.md

```diff
@@ -438,25 +450,139 @@ module.exports = {
-* The `if` property takes an object that works like a MongoDB query criteria object, in a limited way: each property must match the corresponding property of the document. The `$or`, `$and` and `$ne` operators are supported as in MongoDB, along with dot notation to match nested properties. Other MongoDB query features are not supported at this time.
-* The `moduleIf` property works like `if`, but it matches properties of the module rather than the document. Note this extends only to properties passed down to the browser via the `getBrowserData` method of the module in question. You can expose new properties via this method by using the `extendMethods` feature, [as mentioned here](/reference/modules/piece-type.html#getbrowserdata-req).
-* For backward compatibility, this method can also be called with the `moduleName` passed as the first argument and the object as the second, but this is discouraged.
-:::
+Alternatively, you can create a context menu item that emits an event instead of opening a modal:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    label: 'Article',
+    pluralLabel: 'Articles'
+  },
+  init(self) {
+    self.apos.doc.addContextOperation({
+      context: 'update',
+      action: 'refresh-related-content',
+      label: 'Refresh Related Content',
+      type: 'event',
+      if: {
+        type: self.__meta.name
+      }
+    });
+```
+  <template v-slot:caption>
+    modules/article/index.js
+  </template>
+</AposCodeBlock>
+
+
+When a user clicks on an event-type context menu item, it triggers an event with the same name as the `action` property. You can listen for this event elsewhere in your code:
+
+<AposCodeBlock>
+
+``` javascript
+// Listen for the event elsewhere in your code or another module
+self.apos.bus.$on('refresh-related-content', (data) => {
+  // Handle the event
+  // data contains doc, moduleName, moduleLabels, and any props
+  console.log('Refreshing related content for:', data.doc._id);
+
+  // Example: Fetch related content based on tags
+  if (data.doc.tags && data.doc.tags.length) {
+    self.apos.http.get('/api/v1/article', {
+      qs: {
+        tags: { $in: data.doc.tags },
+        _id: { $ne: data.doc._id },
+        limit: 5
+      }
+    }).then(result => {
+      // Update UI or cache with the fresh related content
+      self.apos.notify('Related content refreshed', { type: 'success' });
+    }).catch(err => {
+      self.apos.notify('Failed to refresh related content', { type: 'error' });
+    });
+  }
+});
+```
+</AposCodeBlock>
+
+The event data passed with the event is an object which includes:
+
+```js
+{
+  moduleName: operation.moduleName || this.moduleName,
+  moduleLabels: this.moduleLabels,
+  doc,
+  ...docProps(doc),
```

Comment:
Not sure if it's documents somewhere what this docProps does, it gets some data from the doc. I don't get the point since we pass `doc`.. 🤔.
Maybe we can just get rid of this line since unclear?
edit: May be for some BC stuff here. Let's keep it whatever.